README.md

```diff
@@ -1,26 +1,36 @@
 # Plugin.Firebase
 
-This is a wrapper library around the native Android and iOS Firebase Xamarin SDKs which includes cross-platform APIs for most of the Firebase features. Documentation and the included sample app are MAUI-centric, but the plugin should be usable in any cross-platform .NET6+ project.
+This is a wrapper library around the native Android and iOS Firebase SDKs which includes cross-platform APIs for most of the Firebase features. Documentation and the included sample app are MAUI-centric, but the plugin should be usable in any cross-platform .NET9+ project.
+
+## v4.0 Upgrade Notes
+- The experience of building projects using this plugin has been improved on Windows / Visual Studio. Issues related to hanging builds caused by XamarinBuildDownload and long path issues affecting iOS NuGet packages have been mitigated, if not commpletely resolved.
+- MAUI-specific dependencies have been removed. This makes the plugin friendlier to non-MAUI mobile .NET projects.
+  - Due to this change, it is now necessary to pass in an 'ActivityLocator' function when initializing the plugin on Android. (e.g., `() => Platform.CurrentActivity`)
+- **Minimum** supported .NET version has been increased to .net9.0 (still compatible with .net10.0+)
+- Mimimum supported iOS version has been increased to 15.
+- Minimum supported Android version has been increased to 23.
+- Minimum supported Firebase SDK versions have been increased to 12.5 (iOS) and BoM 33.0 (Android).
+  - Due to this change, consumers should no longer need to explicitly manage AndroidX dependencies on latest MAUI 9 or 10. These could always be needed again in the future as MAUI dependencies shift, or due to conflicts with other plugins etc.
+- Auth provider implementations (Facebook, Apple, Google, Facebook) have been removed. The plugin remains compatible with these (and other) auth providers, but it is now the developers' responsibility to implement them by directly accessing the native SDKs per-platform.
```

Comment:
Also @AdamEssenmacher , since it would be a breaking change for developers, could we point them to some documentation or any sample repository? (e.g. how to add native SDKs to the project?)